src/components/user.js

```diff
@@ -50,34 +97,66 @@ export default class User extends React.Component {
 
             UserAction.updateUser(newUser);
         }
-
         this.setState({ wantEdit: !wantEdit });
         e.preventDefault();
     }
 
-    handleChange (e, type) {
+
+    handleChange (e, obj) {
         const { currentUser, wantEdit } = this.state;
-        const {isNewUser} = this.props;
+        const { isNewUser }             = this.props;
 
         if (wantEdit || isNewUser) {
-            currentUser[type][e.target.name] = e.target.value;
-            this.setState({ currentUser });
+            const name  = e.target.name;
+            const value = e.target.value;
+
+            currentUser[obj][name] = value;
+            this.setState({ currentUser }, () => {
+                this._validateField(name, value);
+            });
         }
         e.preventDefault();
     }
 
-    handleClickSubmit () {
-        const { currentUser } = this.state;
-        const newUser         = new UserInfo(currentUser.info, currentUser.address, currentUser.company);
+    handleClickSubmit (e) {
+        const { currentUser, addDuplicate, formErrors } = this.state;
+        const newUser                                   = new UserInfo(currentUser.info, currentUser.address, currentUser.company);
 
-        UserAction.addNewUser(newUser);
+        if (UserStore.checkUserNameExist(newUser) && !addDuplicate) {
+            this.setState({ addDuplicate: !addDuplicate });
```

Comment:
(proposal)
Can we use `dublicate` to deside what to do instead of using additional `addDuplicate`